app/api/helpers/custom_forms.py

```diff
@@ -33,11 +34,23 @@ def get_schema(form_fields):
 
 
 def validate_custom_form_constraints(form, obj, excluded):
-    form_fields = CustomForms.query.filter_by(
-        form=form,
-        event_id=obj.event_id,
-        is_included=True,
-    ).all()
+    """
+        The validate custom form constraints.
+    """
+    if hasattr(obj, 'ticket_id'):
+        ticket = Ticket.query.filter_by(id=obj.ticket_id).first()
+        form_fields = CustomForms.query.filter_by(
+            form=form,
+            event_id=obj.event_id,
+            is_included=True,
+            form_id=ticket.form_id
+        ).all()
+    else:
+        form_fields = CustomForms.query.filter_by(
+            form=form,
+            event_id=obj.event_id,
+            is_included=True
+        ).all()
```

Comment:
Maybe something in the line of the following
(most probably not correct Python code, though!)
We don't want to duplicate arguments over and over
```
filter_args = ( form=form, event_id=obj.event_id, is_included=True )
if hasattr(obj, 'ticket_id'):
  filter_args += ( form_id=ticket.form_id )
form_fields = CustomForms.query.filter_by( *filter_args ).all()
```

Reply:
Thank you for your suggestion, I have changed it as you want. That's good for us to clean code as well ;)